src/signalrclient/connection_impl.cpp

```diff
@@ -52,6 +54,36 @@ namespace signalr
         }
     }
 
+    connection_impl::connection_impl(const std::string& url, trace_level trace_level, const std::shared_ptr<log_writer>& log_writer,
+        std::shared_ptr<http_client> http_client, std::function<std::shared_ptr<websocket_client>()> websocket_factory)
+        : m_base_url(url), m_connection_state(connection_state::disconnected), m_logger(log_writer, trace_level), m_transport(nullptr),
+        m_message_received([](const std::string&) noexcept {}), m_disconnected([]() noexcept {})
+    {
+        if (http_client != nullptr)
```

Comment:
So the default for logging will be off? This is for connections, so I think it's okay.

Reply:
Not sure what you mean here